trials/multiSurvey.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @description
+ * Builds a multi choice/select survey trial.
+ * @module
+ * @param {Object} options
+ * @param {string} options.responseType - This tells jsPsych which plugin file to use to run the trial. (default: 'survey_multi_choice')
+ * @param {string} options.preamble - HTML formatted string to display at the top of the page above all the questions. (default: empty string)
+ * @param {string|Array} options.prompts - The question prompts, this can be a string (one question) or an Array of strings (multiple questions) (default: "")
+ * @param {Object} options.ansChoices - Object consisting of the key as the answer choice name and value as the array of answer choices. (default: {})
+ */
+
+module.exports = function (options) {
+  const defaults = {
+    responseType: "survey_multi_choice",
+    preamble: "",
+    prompts: [],
+    ansChoices: {},
+  };
+  const { responseType, preamble, prompts, ansChoices } = {
+    ...defaults,
+    ...options,
+  };
+  let questions = [];
+
+  Object.keys(ansChoices).length === 1
+    ? prompts.forEach(function (prompt, index) {
```

Comment:
does this logic need to be split? if the length is 1, wouldn't the second case still work as there would be one item with index 0?

Reply:
I am trying to think of a way to simplify this, because the second case would break for `index=1` onwards as `ansChoices` just has one entry

Reply:
```
 prompts.forEach(function (prompt, index) {
 let ind =Object.keys(ansChoices).length === 1
    ? 0:index
        questions.push({
          prompt: prompt,
          options: ansChoices[Object.keys(ansChoices)[ind]],
          required: true,
        });
      })
```

Reply:
this looks better! thank you @mcmcgrath13